src/status_im/contexts/chat/messenger/composer/effects.cljs

```diff
@@ -1,225 +1,24 @@
-
-(defn use-edit
-  [{:keys [input-ref]}
-   {:keys [text-value saved-cursor-position cursor-position]}
-   {:keys [edit input-with-mentions]}
-   chat-screen-layout-calculations-complete?]
-  (let [mention? (some #(= :mention (first %)) (seq input-with-mentions))]
-    (rn/use-effect
-     (fn []
-       (let [mention-text     (reduce (fn [acc item]
-                                        (str acc (second item)))
-                                      ""
-                                      input-with-mentions)
-             edit-text        (cond
-                                mention? mention-text
-                                ;; NOTE: using text-value for cases when the user
-                                ;; leaves the app with an unfinished edit and re-opens
-                                ;; the chat.
-                                (and (seq @text-value)
-                                     (not (reanimated/get-shared-value
-                                           chat-screen-layout-calculations-complete?)))
-                                @text-value
-                                :else (get-in edit [:content :text]))
-             selection-pos    (count edit-text)
-             inject-edit-text (fn []
-                                (reset! text-value edit-text)
-                                (reset! cursor-position selection-pos)
-                                (reset! saved-cursor-position selection-pos)
-                                (when @input-ref
-                                  (.setNativeProps ^js @input-ref
-                                                   (clj->js {:text edit-text}))))]
-
-         (when (and edit @input-ref)
-           ;; NOTE: A small setTimeout is necessary to ensure the focus is enqueued and is executed
-           ;; ASAP. Check https://github.com/software-mansion/react-native-screens/issues/472
-           ;;
-           ;; The nested setTimeout is necessary to avoid both `on-focus` and
-           ;; `on-content-size-change` handlers triggering the height animation simultaneously, as
-           ;; this causes a jump in the
-           ;; UI. This way, `on-focus` will trigger first without changing the height, after which
-           ;; `on-content-size-change` will animate the height of the input based on the injected
-           ;; text.
-           (js/setTimeout #(do (when (reanimated/get-shared-value
-                                      chat-screen-layout-calculations-complete?)
-                                 (.focus ^js @input-ref))
-                               (reagent/next-tick inject-edit-text))
-                          600))))
-     [(:message-id edit)])))
+     (let [edit-text (get-in edit [:content :text])]
+       (when (and edit @input-ref)
+         (rf/dispatch [:chat.ui/set-chat-input-text edit-text])
+         (.setNativeProps ^js @input-ref #js {:text edit-text})
```

Comment:
Consider `dispatch-sync` if we want to quickly update a value where a text input is subscribed

Reply:
not sure i understand what you mean

Reply:
Just saying that usually for text inputs reading its value from re-frame, we update their value using `dispatch-sync`, otherwise a flicker happens.

Reply:
you mean we should move it to re-frame effect? I'm not sure it worth it, it's already a rn effect so should be fine

Reply:
anyway this code was removed, so no code no problems :)